Source/Core/VideoCommon/Assets/MaterialAsset.cpp

```diff
@@ -329,6 +319,68 @@ bool MaterialData::FromJson(const CustomAssetLibrary::AssetID& asset_id,
+  auto& json_obj = *obj;
+
+  picojson::array json_properties;
+  for (const auto& property : data.properties)
+  {
+    picojson::object json_property;
+    json_property["code_name"] = picojson::value{property.m_code_name};
+
+    std::visit(overloaded{[&](const CustomAssetLibrary::AssetID& value) {
+                            json_property["type"] = picojson::value{"texture_asset"};
+                            json_property["value"] = picojson::value{value};
+                          },
+                          [&](s32 value) {
+                            json_property["type"] = picojson::value{"int"};
+                            json_property["value"] = picojson::value{static_cast<double>(value)};
+                          },
+                          [&](const std::array<s32, 2>& value) {
+                            json_property["type"] = picojson::value{"int2"};
+                            json_property["value"] = picojson::value{ToJsonArray(value)};
+                          },
+                          [&](const std::array<s32, 3>& value) {
+                            json_property["type"] = picojson::value{"int3"};
+                            json_property["value"] = picojson::value{ToJsonArray(value)};
+                          },
+                          [&](const std::array<s32, 4>& value) {
+                            json_property["type"] = picojson::value{"int4"};
+                            json_property["value"] = picojson::value{ToJsonArray(value)};
+                          },
+                          [&](float value) {
+                            json_property["type"] = picojson::value{"float"};
+                            json_property["value"] = picojson::value{static_cast<double>(value)};
+                          },
+                          [&](const std::array<float, 2>& value) {
+                            json_property["type"] = picojson::value{"float2"};
+                            json_property["value"] = picojson::value{ToJsonArray(value)};
+                          },
+                          [&](const std::array<float, 3>& value) {
+                            json_property["type"] = picojson::value{"float3"};
+                            json_property["value"] = picojson::value{ToJsonArray(value)};
+                          },
+                          [&](const std::array<float, 4>& value) {
+                            json_property["type"] = picojson::value{"float4"};
+                            json_property["value"] = picojson::value{ToJsonArray(value)};
+                          },
+                          [&](bool value) {
+                            json_property["type"] = picojson::value{"bool"};
+                            json_property["value"] = picojson::value{value};
+                          }},
+               property.m_value);
+
+    json_properties.emplace_back(std::move(json_property));
+  }
+
+  json_obj["values"] = picojson::value{json_properties};
```

Comment:
```suggestion
  json_obj["values"] = picojson::value{std::move(json_properties)};
```

Since a picojson::array is just a std::vector alias. Missed this in my previous review (my bad)

Reply:
No problem at all.  I didn't consider that.  I'll open another PR for shader as I imagine it can also take advantage of these changes.